src/VisualQueryBuilder/components/LabelFilterItem.tsx

```diff
@@ -95,35 +95,37 @@ export function LabelFilterItem({
             isLoading={state.isLoadingLabelNames}
             options={state.labelNames}
             onChange={(change) => {
-              if (change.label) {
+              if (change.value) {
```

Comment:
I am not sure why we were using here `change.label`, as we always want to use `change.value`. This is not an issue because with label filter options, the label and value have the same content. 

I think there was a confusion here between `label` as Loki label (or other data source label filter's label and `label` which is select's label.